app/routes/questionnaire.py

```diff
@@ -343,6 +342,30 @@ def send_confirmation_email(session_store, schema):
     )
 
 
+@post_submission_blueprint.route("confirmation-email/confirm", methods=["GET", "POST"])
+@with_schema
+@with_session_store
+def confirm_confirmation_email(session_store, schema):
+    try:
+        confirm_email = ConfirmEmail(
+            schema, session_store, request.args["email"], form_data=request.form
+        )
+    except (ConfirmationEmailLimitReached, ConfirmationEmailNotEnabled):
+        return redirect(url_for(".get_thank_you"))
+
+    if request.method == "POST" and confirm_email.form.validate():
+        confirm_email.handle_post()
+        next_location_url = confirm_email.get_next_location_url()
+        return redirect(next_location_url)
```

Comment:
This needs to be similar to IR and need to take into account the `YES` value as it is sending a fulfilment request on every post.

Reply:
Good spot, probably wouldn't go down that path as that naturally starts getting into rendered block. I don't see any issue with using
```
if self._form_data.get("confirm-email") == CONFIRM_EMAIL_YES_VALUE:
```

Reply:
Ya checking form data is fine, IR example was just for info. The value check is done in get_next_location_url which could be removed if the redirect is done in `handle_post` like in IR, but I don't mind what route you go with for that. If it is done in `handle_post` it avoids the double evaluation or alternatively you could use `@cached_property`.

Reply:
You might have missed the line above which mentioned extracted it to a @cached_property

Reply:
Ah ya I did. 👍